### PR TITLE
[No JIRA] Add npm ci to release task

### PR DESCRIPTION
**Pull request by shaundon**
https://github.com/Skyscanner/backpack-android/pull/129/files

### CODE REVIEW

**Comment by k0nserv** [comment]
Isn’t the idea with this that we can’t acciden release with outdated tokens? If so surely we also need to run the gulp build too?

**Comment by shaundon** [comment]
Ah yeah good shout

**Comment by georgegillams** [comment]
To achieve that, we could use a helper like this. Although this will return false if `CHANGELOG.md` has been update, so may need to be tweaked a little.
```
def check_pristine()
  changes = `git status --porcelain`.lines
  result = changes.length < 0
end
```

**Comment by georgegillams** [comment]
This is what a use will see if `gulp` has made changes:
screenshot 2018-11-27 at 09 30 47

**Comment by georgegillams** [comment]
The pristine check allows for changes in `CHANGELOG.md` and `UNRELESED.md` as these are expected when doing a release

File: Rakefile [hunk 1]
```diff
@@ -68,11 +73,13 @@ task ci: [:lint, :analyze, :test]
 
 # task release: :test do
 task release: :ci do
-  sh "npm ci"
   abort red 'Must be on master branch' unless current_branch == 'master'
   abort red 'Must have push access to Backpack on CocoaPods trunk' unless has_trunk_push
   abort red 'Git branch is not up to date please pull' unless branch_up_to_date
 
+  sh "npm ci"
```
Comment (georgegillams) [inline Rakefile]: I moved this as we might as well get the fast checks above done and passed before we make the person releasing wait for `npm ci`

File: Rakefile [hunk 2]
```diff
@@ -51,6 +51,11 @@ def install_pods_in_example_project()
   $?.exitstatus == 0
 end
 
+def check_pristine()
```
Comment (georgegillams) [inline Rakefile]: The pristine check allows for changes in `CHANGELOG.md` and `UNRELESED.md` as these are expected when doing a release

Comment (k0nserv) [reply Rakefile]: Podspec is done after this point right? So it doesn't have to be checked

File: Rakefile [hunk 3]
```diff
@@ -51,6 +51,11 @@ def install_pods_in_example_project()
...
+  changes = `git status --porcelain | grep -Ev "CHANGELOG|UNRELEASED"`.lines
+  result = changes.length < 1
```
Comment (k0nserv) [inline Rakefile]: This can just be `changes.length == 0`

Comment (georgegillams) [reply Rakefile]: Good point

File: Rakefile (hunk 2 above)
Comment (georgegillams) [reply Rakefile]: I don't know what your point is?

Comment (georgegillams) [reply Rakefile]: Do you mean this check should also exclude `Podspec` as it will be checked later anyway?

Comment (k0nserv) [reply Rakefile]: No I think this is fine since `Podspec` is modified later in the flow

**Pull request merged**